llvm/lib/Transforms/IPO/DeadArgumentElimination.cpp

```diff
@@ -820,6 +818,34 @@ bool DeadArgumentEliminationPass::removeDeadStuffFromFunction(Function *F) {
       MDOmitArgs.push_back(AliveArg ? MDOmitArgFalse : MDOmitArgTrue);
     F->setMetadata("sycl_kernel_omit_args",
                    llvm::MDNode::get(F->getContext(), MDOmitArgs));
+    auto FixupMetadata = [&](StringRef MDName) {
```

Comment:
A comment on what this accomplishes can be handy.

Thanks

Reply:
Will add, thanks!